test/form/Form.spec.jsx

```diff
@@ -160,4 +165,41 @@ describe('Form', function () {
       address: {street: 'C'}
     }))
   })
+
+  it('detects pristine state', function () {
+    const {dom: formComponent, firstNameField} = render()
+
+    expect(formComponent, 'to have rendered', <div>pristine true</div>)
+    TestUtils.Simulate.change(firstNameField, {target: {value: 'a'}})
+    expect(formComponent, 'to have rendered', <div>pristine false</div>)
+    TestUtils.Simulate.change(firstNameField, {target: {value: ''}})
+    expect(formComponent, 'to have rendered', <div>pristine true</div>)
+  })
+
+  it('detects submitting state', function () {
+    const {dom: formComponent, form, onSubmit} = render()
+
+    expect(formComponent, 'to have rendered', <div>submitting false</div>)
+    TestUtils.Simulate.submit(form)
+    expect(formComponent, 'to have rendered', <div>submitting false</div>)
+
+    const submit = Bluebird.delay(10)
+    onSubmit.returns(submit)
+    expect(formComponent, 'to have rendered', <div>submitting false</div>)
+    TestUtils.Simulate.submit(form)
+    expect(formComponent, 'to have rendered', <div>submitting true</div>)
+    return submit.then(() => {
+      expect(formComponent, 'to have rendered', <div>submitting false</div>)
+    })
+  })
+
+  it('resets', function () {
+    const {dom: formComponent, firstNameField} = render()
+
+    expect(formComponent, 'to have rendered', <div>pristine true</div>)
+    TestUtils.Simulate.change(firstNameField, {target: {value: 'a'}})
+    expect(formComponent, 'to have rendered', <div>pristine false</div>)
+    formComponent.reset()
+    expect(formComponent, 'to have rendered', <div>pristine true</div>)
```

Comment:
would be nice to check if the firstNameField is empty (since this is the explicit check for the reset)